featuretools/computational_backends/pandas_backend.py

```diff
@@ -151,15 +169,30 @@ def calculate_all_features(self, instance_ids, time_last,
                 if not self.entityset.find_backward_path(start_entity_id=filter_eid,
                                                          goal_entity_id=eid):
                     entity_frames[eid] = eframes_by_filter[eid][eid]
+                    if (large_eframes_by_filter is not None and
+                            eid in large_eframes_by_filter and eid in large_eframes_by_filter[eid]):
+                        large_entity_frames[eid] = large_eframes_by_filter[eid][eid]
 
             if filter_eid in self.feature_tree.ordered_feature_groups:
                 for group in self.feature_tree.ordered_feature_groups[filter_eid]:
                     if verbose:
 
                         pbar.set_postfix({'running': 0})
 
-                    handler = self._feature_type_handler(group[0])
-                    handler(group, entity_frames)
+                    test_feature = group[0]
+
+                    input_frames, output_frames = \
+                        self._determine_input_output_frames(
```

Comment:
seems like this logic can be done in a simpler way to avoid implementing a function that only gets called once. 

```
input_frames = large_entity_frames
if self.feature_tree.needs_all_values_differentiator(test_feature) == 'normal_no_dependent':
    input_frames = entity_frames
```

then you can put the output frame logic in `existing _place_in_output_frames` function you've defined.